cluster/kubernetes/manifests.go

```diff
@@ -7,12 +7,36 @@ import (
 	"github.com/weaveworks/flux/resource"
 )
 
+// namespacer assigns namespaces to manifests that need it (or "" if
+// the manifest should not have a namespace.
+type namespacer interface {
```

Comment:
Wouldn't it be more intuitive to declare the interface in `namespacer.go`?

Reply:
It's declared next to where it's required. This is a bit unfamiliar from a reference point of nominal interface types (in which you assert that your implementation fits the interface, and the compiler tells you if it doesn't).

But Go's interfaces are structural -- you assert that your _value_ fits the interface of the thing you're assigning to, and the compiler checks that you're right. Since there's no need to name the interface when constructing a value, it doesn't need to be accessible outside the package, and since it's used in one place, the most logical places for its definition is next to that.

Granted, what's strictly necessary is not always what's best; and, the rationale above is not exactly applied consistently through this codebase..

Reply:
^ said with the conviction of someone who's only recently adopted this position :-)